toolbox/siestabarriers/SiestaBarriersIO.py

```diff
@@ -0,0 +1,274 @@
+
+#from SiestaBarriers import SiestaBarriersBase
+#from neb_siesta import NEBBase
+#from SiestaBarriers.Utils.utils_siesta import read_siesta_fdf,read_siesta_XV
+#from SiestaBarriers.Utils.utils_vacancy_exchange import pre_prepare_sisl,is_frac_or_cart_or_index,pre_prepare_ase_after_relax
+from .Utils.utils_siesta import read_siesta_fdf,read_siesta_XV
+from .Utils.utils_vacancy_exchange import pre_prepare_sisl,is_frac_or_cart_or_index,pre_prepare_ase_after_relax
+
+import os
+import glob,shutil
+#from SiestaBarriers.Utils.utils_io import Ghost_block , file_cat,string_cat
+#from SiestaBarriers.Utils.utils_analysis import NEB_Barrier
+#from SiestaBarriers.Utils.utils_io import replace
+#from SiestaBarriers.Utils.utils_analysis import NEB_Fix
+
+from .Utils.utils_io import Ghost_block , file_cat,string_cat
+from .Utils.utils_analysis import NEB_Barrier
+from .Utils.utils_io import replace
+from .Utils.utils_analysis import NEB_Fix
+
+
+__author__ = "Arsalan Akhatar"
+__copyright__ = "Copyright 2021, SIESTA Group"
+__version__ = "0.1"
+__maintainer__ = "Arsalan Akhtar"
+__email__ = "arsalan_akhtar@outlook.com," + \
+        " miguel.pruneda@icn2.cat "
+__status__ = "Development"
+__date__ = "Janurary 30, 2021"
+
+
+class SiestaBarriersIO():
+    """
+    """
+    def __init__(self,
+                 sisl_images,
+                 flos_path ,
+                 flos_file_name_relax,
+                 flos_file_name_neb,
+                 number_of_images,
+                 initial_relaxed_path,
+                 final_relaxed_path,
+                 relax_engine,
+                 relaxed,
+                 ghost,
+                 initial_relaxed_fdf_name,
+                 final_relaxed_fdf_name,
+                 neb_results_path = None,
+                 ):
+        
+        self.sisl_images = sisl_images
+        self.flos_path = flos_path
+        self.flos_file_name_relax = flos_file_name_relax
+        self.flos_file_name_neb = flos_file_name_neb
+        self.number_of_images = number_of_images
+        self.initial_relaxed_path = initial_relaxed_path 
+        self.final_relaxed_path = final_relaxed_path
+        self.relax_engine = relax_engine
+        self.relaxed = relaxed 
+        self.ghost = ghost
+        self.initial_relaxed_fdf_name = initial_relaxed_fdf_name
+        self.final_relaxed_fdf_name = final_relaxed_fdf_name
+        self.neb_results_path = neb_results_path
+
+    def write_all_images_sisl(self, fname = 'images',folder_name = 'all' , out_format = 'xyz'):
+        """
+        """
+        if self.relaxed == True:
+            print ("Generating All Images After Relaxation")
+            folder_name = folder_name + '_relaxed'
+            if os.path.isdir(folder_name):
+                print (" The All Image Folder Relaxed is there Already PASS")
+                print (" Check The Folder: '{}' ".format(folder_name))
+                pass
+            else:
+                os.mkdir(folder_name)
```

Comment:
If you can, please use `pathlib` instead, some operations exists there which are easier to use. Some don't, in which case you do need to use `os`.